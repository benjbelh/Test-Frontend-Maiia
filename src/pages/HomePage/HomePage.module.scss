
.HomePageContainer {
    background: url(../../assets/img/e-commerce-backgroud.jpg) no-repeat center fixed;
    -webkit-background-size: cover;
    background-size: cover;
    height: 100vh;
    
    .HomePageContent {
        display: flex;
        flex-direction: column;
        align-items: flex-end;
        justify-content: center;
        max-width: 1400px;
        margin: 0 auto;
        height: 100%;
        padding-right: 16px;

        .HomePageTitle {
            font-family: 'Concert One', cursive;
            font-size: clamp(2.2rem,6vw,5rem);
            text-align: right;
        }
    } 
}


