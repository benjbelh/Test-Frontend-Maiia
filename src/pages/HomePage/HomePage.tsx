import { Button } from '@material-ui/core';
import { KeyboardArrowRight } from '@material-ui/icons';
import React, { FC, useCallback } from 'react';
import { useHistory } from 'react-router-dom';
import styles from './HomePage.module.scss';

const HomePage: FC<{}> = () => {
  const history = useHistory();

  const onButtonClick = useCallback(() => {
    history.push(`/products`);
  }, [history]);

  return (
    <div className={styles.HomePageContainer}>
      <div className={styles.HomePageContent}>
        <div className={styles.HomePageTitle}>
          <p className={styles.HomePageTitle}>
            OFFRONS-NOUS
            <br />
            LES PRODUITS
            <br />
            DE NOS RÃŠVES
          </p>
        </div>
        <Button
          variant="contained"
          endIcon={<KeyboardArrowRight />}
          size="large"
          onClick={onButtonClick}
          data-testid="product-page-access"
        >
          ACCEDER AU CATALOGUE
        </Button>
      </div>
    </div>
  );
};
export default HomePage;
