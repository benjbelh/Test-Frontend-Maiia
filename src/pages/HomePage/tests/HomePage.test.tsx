import React from 'react';
import { render } from '@testing-library/react';
import HomePage from '../HomePage';

describe('HomePage', () => {
  it('should renders correctly', () => {
    const { asFragment } = render(<HomePage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
