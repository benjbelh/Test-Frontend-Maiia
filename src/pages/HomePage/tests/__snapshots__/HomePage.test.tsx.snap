// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should renders correctly 1`] = `
<DocumentFragment>
  <div
    class="HomePageContainer"
  >
    <div
      class="HomePageContent"
    >
      <div
        class="HomePageTitle"
      >
        <p
          class="HomePageTitle"
        >
          OFFRONS-NOUS
          <br />
          LES PRODUITS
          <br />
          DE NOS RÃŠVES
        </p>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge"
        data-testid="product-page-access"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          ACCEDER AU CATALOGUE
          <span
            class="MuiButton-endIcon MuiButton-iconSizeLarge"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
